File: Trunk/IAS/CommonFeature/Session/XASessionUtility/SessionUtility.cpp
// This is the main DLL file.

#include "McsfDataHeader/mcsf_data_header_common.h"
#include "SessionUtility.h"

using namespace std;
MCSF_DATAHEADER_BEGIN_NAMESPACE 

string CSessionUtility::SerializeDataHead(IDataHeaderElementMap* spDataHeaderElemMap)
 {
	 string sSerializedOutput("");
	 IDataHeaderElementMap::Serialize(spDataHeaderElemMap,&sSerializedOutput);
	 return sSerializedOutput;
 }

void CSessionUtility::DeserializeDataHead(string serializedDataHead, IDataHeaderElementMap* pHeader)
{
	IDataHeaderElementMap::Deserialize(serializedDataHead,pHeader);
}

CommandContext* CSessionUtility::CreateCommandContext(string receiver, int commandID, string serializeObject)
{
	Mcsf::CommandContext* context = new Mcsf::CommandContext();
	context->sReceiver = receiver;
	context->iCommandId = commandID;
	context->sSerializeObject = serializeObject;
	return context;
}

string CSessionUtility::RequestCommandProtoBuffer(int commandID, vector<string>&pVector )
{
	UIH::XR::Basis::Proto::SessionRequestCommand sRequestCommand;
	sRequestCommand.set_commandid(commandID);

	 for (vector<string>::iterator  iter =pVector.begin(); iter != pVector.end(); iter++)
    {
		std::string str = *iter;
		sRequestCommand.add_parameters(str);
    }
	string sRequestCommandString("");
	sRequestCommand.SerializeToString(&sRequestCommandString);
	return sRequestCommandString;
}

MCSF_DATAHEADER_END_NAMESPACE

File: Trunk/IAS/CommonFeature/Session/UIH.XA.SessionUtility/SessionUtility.cs
using UIH.Mcsf.Pipeline.Data;
using UIH.Mcsf.Core;
using UIH.XR.Basis.Proto;
using Google.ProtocolBuffers;

namespace UIH.XA.SessionManager
{
    public class SessionUtility
    {
        /// <summary>
        /// Serialize dataHead
        /// </summary>
        public static byte[] SerializeDataHead(DicomAttributeCollection dataHead)
        {
            byte[] bValue = null;
            bool result = dataHead.Serialize(out bValue);
            return bValue;
        }

        /// <summary>
        /// Deserialize DataHead
        /// </summary>
        public static DicomAttributeCollection DeserializeDataHead(byte[] serializedDataHead)
        {
            return DicomAttributeCollection.Deserialize(serializedDataHead);
        }

        /// <summary>
        /// Create commandContext
        /// </summary>
        /// <param name="receiver">receiver of communication</param>
        /// <param name="commandID">commandID of communication</param>
        /// <param name="serializeObject">message needs to send</param>
        /// <returns>CommandContext object</returns>
        public static CommandContext CreateCommandContext(string receiver, int commandID, byte[] serializeObject)
        {
            CommandContext context = new CommandContext();
            context.sReceiver = receiver;
            context.iCommandId = commandID;
            context.sSerializeObject = serializeObject;
            return context;
        }

        /// <summary>
        /// Create a protobuffer object
        /// </summary>
        /// <param name="commandID">cmd</param>
        /// <param name="parArray">parameter array</param>
        /// <returns>protobuffer</returns>
        public static byte[] RequestCommandProtoBuffer(int commandID,ByteString[] parArray)
        {
            SessionRequestCommand.Builder builder = new SessionRequestCommand.Builder();
            builder.SetCommandId(commandID);
            builder.AddRangeParameters(parArray);
            return builder.Build().ToByteArray();
        }
    }
}
